tests/modules.py

```diff
@@ -108,6 +133,49 @@ def test_module_import(network, args):
     return network
 
 
+@reqs.description("Test Node.js/npm app")
+def test_npm_app(network, args):
+    primary, _ = network.find_nodes()
+
+    LOG.info("Building npm app")
+    app_dir = os.path.join(THIS_DIR, "npm-app")
+    subprocess.run(["npm", "ci"], cwd=app_dir, check=True)
+    subprocess.run(["npm", "run", "build"], cwd=app_dir, check=True)
+
+    LOG.info("Deploying npm app modules")
+    kv_prefix = "/my-npm-app"
+    dist_dir = os.path.join(app_dir, "dist")
+    for module_path in glob.glob(os.path.join(dist_dir, "**", "*.js"), recursive=True):
+        module_name = os.path.join(kv_prefix, os.path.relpath(module_path, dist_dir))
+        proposal_body, _ = ccf.proposal_generator.set_module(module_name, module_path)
+        proposal = network.consortium.get_any_active_member().propose(
+            primary, proposal_body
+        )
+        network.consortium.vote_using_majority(primary, proposal)
+
+    LOG.info("Deploying endpoint script")
+    with tempfile.NamedTemporaryFile("w") as f:
+        f.write(NPM_APP_SCRIPT)
+        f.flush()
+        network.consortium.set_js_app(remote_node=primary, app_script_path=f.name)
+
+    LOG.info("Calling npm app endpoints")
+    with primary.client("user0") as c:
+        body = [1, 2, 3, 4]
+        r = c.post("/app/npm/partition", body)
+        assert r.status_code == 200, r.status_code
+        assert json.loads(r.body) == [[1, 3], [2, 4]], r.body
+
+        r = c.post("/app/npm/pb", body)
+        assert r.status_code == 200, r.status_code
+        assert len(r.body) > 0, r.body
```

Comment:
What's this call actually doing? This seems like a very weak assertion, can we add something more precise?

Reply:
I changed it to be more strong. This endpoint is wrapping the input data in a simple protobuf message with an ad-hoc created schema.